### ISSUE TITLE
CSharp Client Generator - Not returning result nor throwing exception for non-success HTTP status codes if IsPlainText = true

### ISSUE THREAD

**Issue by francozuca** [RicoSuter/NSwag#3512]
Hi, 

I'm having an issue trying generate a C# client out of an OpenAPI spec which contains plain/text responses.

Specifically, I have this error telling me that **not all code paths return a value** if the operations have **non-success HTTP status codes** as you can see below:
![image](https://user-images.githubusercontent.com/66265246/124018219-007ae600-d9be-11eb-88d6-fe6f1d0caafd.png)

I think that the problem happens here:
https://github.com/RicoSuter/NSwag/blob/1c836ee3cc766ce25f3170618a5d12e5e96fde61/src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.ProcessResponse.liquid#L15

Basically we will need to **add an else block** to deal with that non-success HTTP codes and throw an exception properly.

Does that make sense? If it does I can fix this and raise a PR.
Thanks!

**Comment by icnocop** [comment]
As a work-around, I have to downgrade the nuget package "NSwag.ApiDescription.Client" to version "13.6.2".

This is an issue with the nuget package "NSwag.ApiDescription.Client" version "13.7.0" and higher.

**Comment by RicoSuter** [comment]
So it's currently producing code which does not compile?

**Comment by francozuca** [comment]
> So it's currently producing code which does not compile?

@RicoSuter, exactly.

**Comment by RicoSuter** [comment]
And just a regular 

    return result_;

would be enough?

**Comment by francozuca** [comment]
Either that, or throw an exception as we are doing for the "response to object deserialization case":
![image](https://user-images.githubusercontent.com/66265246/124018414-3324de80-d9be-11eb-8f7e-64ae2d2eca6e.png)

**Comment by RicoSuter** [comment]
I think this should fix it: 
https://github.com/RicoSuter/NSwag/pull/3530
What do you think?

**Comment by RicoSuter** [comment]
Super thx for checking. 
Is this fix urgent for you? For now you could also use the new template file and replace it via the replacement directory option...

**Comment by francozuca** [comment]
@RicoSuter no urgency at all, actually I did exactly that as a workaround :) I'll remove my template once the new version is released.

Thanks!

**Comment by icnocop** [comment]
Hi.

I tried to use the updated template file `Client.Class.ProcessResponse.liquid` but I get the following compiler error:

```
error CS0103: The name 'objectResponse_' does not exist in the current context
```

I'm using the `NSwag.ApiDescription.Client` nuget package version 13.11.3.

```
    <OpenApiReference Include="ApiClient.json" CodeGenerator="NSwagCSharp" Namespace="MyNamespace.ApiClient" ClassName="ApiClient" Options="/ClientBaseClass:ApiClientBase /UseHttpRequestMessageCreationMethod:true /TemplateDirectory:template">
      <SourceUri>http://localhost/MyWebApplication/docs/v1/swagger.json</SourceUri>
      <OutputPath>..\ApiClient.cs</OutputPath>
    </OpenApiReference>
```

Any ideas?

Thank you.
